src/test/java/net/sf/jabref/FileBasedTestCase.java

```diff
@@ -38,78 +39,55 @@ public void setUp() throws IOException {
-        Assert.assertTrue(pdf1a.createNewFile());
+        Files.createDirectory(rootDir.resolve("2002"));
+        Path dir2003 = Files.createDirectory(rootDir.resolve("2003"));
+        assertTrue(Files.exists(Files.createFile(dir2003.resolve("Paper by HipKro03.pdf"))));
 
-        File subDir2 = new File(root, "pdfs");
-        Assert.assertTrue(subDir2.mkdir());
+        Path dirTest = Files.createDirectory(rootDir.resolve("test"));
+        assertTrue(Files.exists(Files.createFile(dirTest.resolve(".TEST"))));
+        assertTrue(Files.exists(Files.createFile(dirTest.resolve("TEST["))));
+        assertTrue(Files.exists(Files.createFile(dirTest.resolve("TE.ST"))));
+        assertTrue(Files.exists(Files.createFile(dirTest.resolve("foo.dat"))));
 
-        File subsubDir1 = new File(subDir2, "sub");
-        Assert.assertTrue(subsubDir1.mkdir());
+        Path graphicsDir = Files.createDirectory(rootDir.resolve("graphicsDir"));
+        Path graphicsSubDir = Files.createDirectories(graphicsDir.resolve("subDir"));
 
-        File pdf2 = new File(subsubDir1, "HipKro03-sub.pdf");
-        Assert.assertTrue(pdf2.createNewFile());
-
-        File dir2002 = new File(root, "2002");
-        Assert.assertTrue(dir2002.mkdir());
-
-        File dir2003 = new File(root, "2003");
-        Assert.assertTrue(dir2003.mkdir());
-
-        File pdf3 = new File(dir2003, "Paper by HipKro03.pdf");
-        Assert.assertTrue(pdf3.createNewFile());
-
-        File dirTest = new File(root, "test");
-        Assert.assertTrue(dirTest.mkdir());
-
-        File pdf4 = new File(dirTest, "HipKro03.pdf");
-        Assert.assertTrue(pdf4.createNewFile());
-
-        File pdf5 = new File(dirTest, ".TEST");
-        Assert.assertTrue(pdf5.createNewFile());
-
-        File pdf6 = new File(dirTest, "TEST[");
-        Assert.assertTrue(pdf6.createNewFile());
-
-        File pdf7 = new File(dirTest, "TE.ST");
-        Assert.assertTrue(pdf7.createNewFile());
-
-        File foo = new File(dirTest, "foo.dat");
-        Assert.assertTrue(foo.createNewFile());
-
-        File graphicsDir = new File(root, "graphicsDir");
-        Assert.assertTrue(graphicsDir.mkdir());
-
-        File graphicsSubDir = new File(graphicsDir, "subDir");
-        Assert.assertTrue(graphicsSubDir.mkdir());
-
-        File jpg = new File(graphicsSubDir, "HipKro03test.jpg");
-        Assert.assertTrue(jpg.createNewFile());
-
-        File png = new File(graphicsSubDir, "HipKro03test.png");
-        Assert.assertTrue(png.createNewFile());
+        assertTrue(Files.exists(Files.createFile(graphicsSubDir.resolve("HipKro03test.jpg"))));
+        assertTrue(Files.exists(Files.createFile(graphicsSubDir.resolve("HipKro03test.png"))));
 
     }
 
     @After
     public void tearDown() {
-        FileBasedTestHelper.deleteRecursive(root);
+
+        try {
+            for (Path p : Files.walk(rootDir).sorted((a, b) -> b.compareTo(a)). // reverse; files before dirs
```

Comment:
Files.walk creates a stream, which must be closed using try-with-resources, otherwise, this may affect another test as it will not free the directory resource.